src/Foundatio.RabbitMQ/Messaging/RabbitMQMessageService.cs

```diff
@@ -178,7 +198,11 @@ public class RabbitMQMessageService : MessageBusBase, IMessageBus {
             if (_subscriberClient != null && _subscriberClient.IsOpen) {
                 _subscriberClient.Close();
             }
-            _publisherChannel?.Abort();
+            if (_lock != null) {
+                lock (_lock) {
+                    _publisherChannel?.Abort();
+                }
+            }
             _subscriberChannel?.Abort();
```

Comment:
@niemyjski  In the RabbitMQ documentation, it clearly says that IModel instances should not be used by more than one thread simoultaneously. Logically publish message is the one that can be used across mulitple threads, however, when it comes to subscription we do that only once . So couldnt think of any scenario where we will call subscribe many times from the same or multiple threads. If its across process then ofcourse the RabbitMQMessageService intance will be different.  Atleast that was my thinking :)